beat/heart.py

```diff
@@ -3281,11 +3355,24 @@ def gamma(takeoffangles, azimuths):
     sa = num.sin(azimuths)
     return num.array([st*ca, st*sa, ct])
 
+def symmat6(a_xx, a_yy, a_zz, a_xy, a_xz, a_yz):
+    '''
+    modified from pyrocko
+    Create symmetric 3x3 matrix from its 6 non-redundant values.
+    '''
+    return num.matrix([[a_xx, a_xy, a_xz],
+                       [a_xy, a_yy, a_yz],
+                       [a_xz, a_yz, a_zz]], dtype=num.float)
+
 def pol_synthetics(sources, targets):  
     takeoffangles = [target.take_offangle_rad for target in targets]
     azimuths = [target.azimuth_rad for target in targets]
     gamma_co = gamma(takeoffangles, azimuths)
-    amps = num.array([num.diag(gamma_co.T.dot(sources[0].m9/sources[0].moment).dot(gamma_co))]).T
+    moment = sources[0].pyrocko_moment_tensor()
+    m9 = moment.m()
```

Comment:
this normalisation for example is handled already at the beat.sources.MTSourceWithMagnitude

Reply:
and it shouldnt be needed for the MTQTSource e.g. ... can you resolve magnitude with polarities anyways?

Reply:
> and it shouldnt be needed for the MTQTSource e.g. ... can you resolve magnitude with polarities anyways?

Actually, in all results, the magnitude was closely resolved, I was amazed.

Reply:
> this normalisation for example is handled already at the beat.sources.MTSourceWithMagnitude

you considered different classes for sources in BEAT, but unfortunately they don't have the identical attributes or function to extract moment tensor itself. Tried different ways, from what I realized, it's the only way. 
for MTQT, source.m9() worked,
for MT, m() worked, and for DCSource, none of them, so that's why I used pyrocko_moment_tensor()
for some types of source, you defined normalized moment tensor, for theoretical amplitude we need normalized moment.